scylla-cql/src/errors.rs

```diff
@@ -538,6 +538,23 @@ pub enum RequestError {
     UnableToAllocStreamId,
 }
 
+impl From<RequestError> for UserRequestError {
+    fn from(value: RequestError) -> Self {
+        match value {
```

Comment:
Ah, no. It's not a subset - it's an interpretation in a different context. The key is that it maps some variants proper to non-user requests to `UnexpectedResponse`. Got it right?

Reply:
> Ah, no. It's not a subset - it's an interpretation in a different context. The key is that it maps some variants proper to non-user requests to `UnexpectedResponse`. Got it right?

Exactly. That's something we need to do, because we don't have the request's context when creating a `CqlResponseParseError` (in Response::deserialize).